arch/arc/core/irq_offload.c

```diff
@@ -36,13 +35,13 @@ static void arc_irq_offload_handler(const void *unused)
 {
 	ARG_UNUSED(unused);
 
-	offload_params[CURR_CPU].fn(offload_params[CURR_CPU].arg);
+	offload_params[_current_cpu->id].fn(offload_params[_current_cpu->id].arg);
 }
 
 void arch_irq_offload(irq_offload_routine_t routine, const void *parameter)
 {
-	offload_params[CURR_CPU].fn = routine;
-	offload_params[CURR_CPU].arg = parameter;
+	offload_params[_current_cpu->id].fn = routine;
+	offload_params[_current_cpu->id].arg = parameter;
```

Comment:
Hmmm, now this require to be guarded by interrupt lock - otherwise we'll get assertion on SMP platform (due to _current_cpu macro).

Reply:
Yep